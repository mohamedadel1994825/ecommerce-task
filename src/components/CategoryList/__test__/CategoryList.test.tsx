import { cleanup, render } from "@testing-library/react-native";
import React from "react";
import { CategoryList } from "../index";
jest.mock("../../../store/productsData.json", () => ({
  categories: [
    { id: 1, name: "Category 1", brands: [{ id: 1, name: "Brand 1" }] },
    { id: 2, name: "Category 2", brands: [{ id: 2, name: "Brand 2" }] },
    { id: 3, name: "Category 3", brands: [{ id: 3, name: "Brand 3" }] },
  ],
}));
describe("CategoryList", () => {
  afterEach(cleanup);

  const selectedCategory = {
    id: 2,
    name: "Category 2",
    brands: [{ id: 2, name: "Brand 2" }],
  };

  const setSelectedCategory = jest.fn();
  const setSelectedBrand = jest.fn();

  it("should match snapshot", () => {
    const { toJSON } = render(
      <CategoryList
        selectedCategory={selectedCategory}
        setSelectedCategory={setSelectedCategory}
        setSelectedBrand={setSelectedBrand}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
