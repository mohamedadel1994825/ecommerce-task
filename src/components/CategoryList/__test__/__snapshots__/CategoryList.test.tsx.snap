// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should match snapshot 1`] = `
[
  <Text
    style={
      {
        "fontSize": 15,
        "fontWeight": "500",
        "marginBottom": 14,
        "marginTop": 10,
        "paddingLeft": 15,
      }
    }
  >
    navigation:home:categories
  </Text>,
  <View>
    <RCTScrollView
      ItemSeparatorComponent={
        {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      contentContainerStyle={
        {
          "paddingHorizontal": 15,
        }
      }
      data={
        {
          "categories": [
            {
              "brands": [
                {
                  "id": 1,
                  "name": "Brand 1",
                },
              ],
              "id": 1,
              "name": "Category 1",
            },
            {
              "brands": [
                {
                  "id": 2,
                  "name": "Brand 2",
                },
              ],
              "id": 2,
              "name": "Category 2",
            },
            {
              "brands": [
                {
                  "id": 3,
                  "name": "Brand 3",
                },
              ],
              "id": 3,
              "name": "Category 3",
            },
          ],
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>,
  <View
    style={
      {
        "backgroundColor": "#efefef",
        "height": 1,
        "marginBottom": 20,
        "width": "100%",
      }
    }
  />,
]
`;
